src/main/kotlin/org/opensearch/replication/action/index/TransportReplicateIndexClusterManagerNodeAction.kt

```diff
@@ -127,7 +127,8 @@ class TransportReplicateIndexClusterManagerNodeAction @Inject constructor(transp
                 persistentTasksService.waitForTaskCondition(task.id, replicateIndexReq.timeout()) { t ->
                     val replicationState = (t.state as IndexReplicationState?)?.state
                     replicationState == ReplicationState.FOLLOWING ||
-                            (!replicateIndexReq.waitForRestore && replicationState == ReplicationState.RESTORING)
+                            (!replicateIndexReq.waitForRestore && replicationState == ReplicationState.RESTORING) ||
+                            (!replicateIndexReq. waitForRestore && replicationState == ReplicationState. FAILED)
```

Comment:
Does handling for FAILED exist?

(Also a nit - there is a space after the `.`)

Does this need any update to test case?

Reply:
ohh yes thanks for pointing out the space after ```.```